kb(treeview): Expand and collapse items onclick (#1201)
telerik/blazor-docs: knowledge-base/treeview-expand-collapse-on-item-click.md

---
title: Expand and collapse TreeView items on click
description: How to expand and collapse the parent TreeView items when clicking on them?
type: how-to
page_title: Expand and collapse TreeView items on click
slug: treeview-kb-expand-collapse-on-item-click
position: 
tags: treeview, item, expand, collapse, click, onclick
ticketid: 1583333
res_type: kb
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>TreeView for Blazor</td>
		</tr>
	</tbody>
</table>

## Description

Currently, I am able to expand or collapse the parent TreeView items only when I click on their arrow buttons.

How can I expand the items when I click on their text? Is it also possible to collapse an item when I click it?

## Solution

By design, the parent TreeView items are expanded or collapsed on click of the expand/collapse icon that is rendered next to the item.

To expand a TreeView item when you click on it:

* Handle the [`OnItemClick`]({%slug treeview-events%}#onitemclick) event.

> * Verify if the clicked item is a parent one to ensure it can be expanded:

Reviewer: ```suggestion
* Verify if the clicked item is a parent, so it can be expanded:
```

> 	* For [flat data]({%slug components/treeview/data-binding/flat-data%}) check if `hasChildren` field equals `true`.

Reviewer: ```suggestion
	* For [flat data]({%slug components/treeview/data-binding/flat-data%}) check if `HasChildren` field equals `true`.
```
